### ISSUE TITLE
`--json` & `--debug` command line options ignored

### ISSUE THREAD

**Issue by jaimergp** [conda/conda#12928]
### What happened?

It looks like conda 23.7.0 does not respect the `--json` flag, in any command as far as I can tell:

conda 23.5.2 would output:

```
$ conda info --json       
{
  "GID": 20,
  "UID": 501,
  "active_prefix": "/Users/jrodriguez/.local/anaconda",
  "active_prefix_name": "base",
  "av_data_dir": "/Users/jrodriguez/.local/anaconda/etc/conda",
  "av_metadata_url_base": null,
  "channels": [
    "https://conda.anaconda.org/conda-forge/osx-arm64",
    "https://conda.anaconda.org/conda-forge/noarch"
  ],
  "conda_build_version": "3.25.0",
  "conda_env_version": "23.5.2",
  "conda_location": "/Users/jrodriguez/.local/anaconda/lib/python3.9/site-packages/conda",
  "conda_prefix": "/Users/jrodriguez/.local/anaconda",
  "conda_shlvl": 3,
  "conda_version": "23.5.2",
  "config_files": [
    "/Users/jrodriguez/.local/anaconda/.condarc",
    "/Users/jrodriguez/.condarc"
  ],
  "default_prefix": "/Users/jrodriguez/.local/anaconda",
... more hidden
```

But 23.7.0 just defaults to plain text despite the flag:

```
$ conda info --json

     active environment : base
    active env location : /Users/jrodriguez/.local/anaconda
            shell level : 3
       user config file : /Users/jrodriguez/.condarc
 populated config files : /Users/jrodriguez/.local/anaconda/.condarc
                          /Users/jrodriguez/.condarc
          conda version : 23.7.0
    conda-build version : 3.25.0
         python version : 3.9.16.final.0
       virtual packages : __archspec=1=arm64
                          __osx=13.3=0
                          __unix=0=0
       base environment : /Users/jrodriguez/.local/anaconda  (writable)
      conda av data dir : /Users/jrodriguez/.local/anaconda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/jrodriguez/.local/anaconda/pkgs
                          /Users/jrodriguez/.conda/pkgs
       envs directories : /Users/jrodriguez/.local/anaconda/envs
                          /Users/jrodriguez/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.7.0 requests/2.28.2 CPython/3.9.16 Darwin/22.4.0 OSX/13.3 solver/libmamba conda-libmamba-solver/23.5.0 libmambapy/1.4.1
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

### Conda Details

<details>
<summary><code>conda config</code></summary>

```shell
==> /Users/jrodriguez/.local/anaconda/.condarc <==
channels:
  - conda-forge

==> /Users/jrodriguez/.condarc <==
add_pip_as_python_dependency: False
auto_update_conda: False
local_repodata_ttl: 0
channel_priority: strict
channels:
  - conda-forge
show_channel_urls: True
experimental:
  - jlap
solver: libmamba
```

</details>

### Additional Context

This has broken some conda-forge infra depending on programmatic JSON usage of `conda`, as well conda-libmamba-solver's CI.

**Comment by kenodegard** [comment]
Resolved in #12929 and released with [conda 23.7.1](https://github.com/conda/conda/releases/tag/23.7.1)

**Comment by jaimergp** [comment]
Confirmed it fixes the conda-libmamba-solver failures! Thanks!
